src/modalities/registry/components.py

```diff
@@ -106,7 +118,41 @@ class ComponentEntity:
     ),
     ComponentEntity("model", "checkpointed", ModelFactory.get_checkpointed_model, CheckpointedModelConfig),
     ComponentEntity("model", "fsdp_wrapped", ModelFactory.get_fsdp_wrapped_model, FSDPWrappedModelConfig),
+    ComponentEntity(
+        "model", "weight_initialized", ModelFactory.get_weight_initalized_model, WeightInitializedModelConfig
+    ),
     ComponentEntity("model", "coca", CoCa, CoCaConfig),
+    # weight initializers
+    ComponentEntity(
+        "weight_initialization",
+        "composed",
+        HighLevelWeightInitializationFactory.get_composed_weight_init,
+        ComposedWeightInitializationConfig,
+    ),
+    ComponentEntity(
+        "weight_initialization",
+        "wrapper",
+        LowLevelInitializationFactory.get_weight_initializer_wrapper,
+        WeightInitializerWrapperConfig,
+    ),
+    ComponentEntity(
+        "weight_initialization",
+        "plain",
+        LowLevelInitializationFactory.get_plain_initialization,
+        PlainWeightInitializationConfig,
```

Comment:
```suggestion
        PlainInitializationConfig,
```
class name would have to be changed too ofc